src/wallet/wallet.cpp

```diff
@@ -35,6 +35,8 @@
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/thread.hpp>
 
+/** Protects the vpwallets vector */
+CCriticalSection cs_wallets;
 std::vector<CWalletRef> vpwallets;
```

Comment:
Why? `vpwallets` is accessed by functions in several of the wallet source files.